### PR TITLE
Adds comprehensive test coverage for UPE checkouts

### LINKED ISSUE #2016: UPE - Unit Tests: Regular Checkout Edge Cases
**Is your feature request related to a problem? Please describe.**
A journey of 100% test coverage begins with a single assertion. This is one small step towards complete unit test coverage for the UPE gateway and its recently created dependencies.

**Describe the solution you'd like**
Specifically the following functions on `WC_Stripe_UPE_Payment_Gateway` could use some more comprehensive unit test coverage.
 - `maybe_process_upe_redirect`
 - `process_upe_redirect_payment`
 - `process_order_for_confirmed_intent`

Additionally, I think we could use a little more coverage on errors during checkout.

**Describe alternatives you've considered**
Inserting fingers into ears and going "Lalalalalalala..."

**Additional context**
Related issues: #2014, #2015

### PR DESCRIPTION
**Pull request by FangedParakeet**
# Changes proposed in this Pull Request:

Issue: Link to the GitHub issue this PR addresses (if appropriate).
Fixes #2016 

This PR adds more comprehensive test coverage for all regular flows during UPE checkouts (excluding any that are subscription related or checkouts using a saved payment method--which are already comprehensively covered).

# Testing instructions

Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?

1. Pull the changes in this PR and start up your local environment (`npm run up`).
2. Run the tests (`npm run test:php`).
3. Hopefully, see green dots all across the board.
4. Celebrate!

-------------------
- [ ] Make sure your changes respect [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- [ ] Did you make changes, or create a **new .js file**? If **Gruntfile.js** exists in the repo, make sure to run `grunt`.

### CODE REVIEW

File: includes/payment-methods/class-wc-stripe-upe-payment-gateway.php [hunk 1]
```diff
@@ -1045,7 +1045,7 @@ public function set_payment_method_title_for_order( $order, $payment_method_type
 	 * @return string
 	 */
 	public function generate_upe_checkout_experience_accepted_payments_html( $key, $data ) {
-		$stripe_account      = WC_Stripe_API::retrieve( 'account' );
+		$stripe_account      = $this->stripe_request( 'account' );
 		$stripe_capabilities = isset( $stripe_account->capabilities ) ? (array) $stripe_account->capabilities : [];
```
Comment (FangedParakeet) [inline includes/payment-methods/class-wc-stripe-upe-payment-gateway.php:1049]: I really think this should be handled using the `WC_Stripe_Account` class, which is initialised and stored as a property of `WC_Stripe`, since this class has better error handling and stores the response in a transient...but that's going to have to be a story for another day.

File: tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php [hunk 2]
```diff
@@ -186,11 +203,11 @@ public function test_process_redirect_payment_returns_valid_response() {
 		$payment_method_id = 'pm_mock';
 		$customer_id       = 'cus_mock';
 		$order             = WC_Helper_Order::create_order();
-		$currency          = $order->get_currency();
 		$order_id          = $order->get_id();
 
 		list( $amount, $description, $metadata ) = $this->get_order_details( $order );
 		$order->set_payment_method( WC_Stripe_UPE_Payment_Gateway::ID );
+		$order->save();
```
Comment (FangedParakeet) [inline tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php:210]: These tests still somehow worked without this line, because the mock gateway also calls `$order->save()`; but honestly, this should be here, because nothing really gets stored to the order unless we save it.

File: tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php [hunk 3]
```diff
@@ -226,7 +243,7 @@ public function test_process_redirect_payment_returns_valid_response() {
 		$this->assertEquals( 'processing', $final_order->get_status() );
 		$this->assertEquals( 'Credit card / debit card', $final_order->get_payment_method_title() );
 		$this->assertEquals( $payment_intent_id, $final_order->get_meta( '_stripe_intent_id', true ) );
-		$this->assertEquals( true, $final_order->get_meta( '_stripe_upe_redirect_processed', true ) );
+		$this->assertTrue( (bool) $final_order->get_meta( '_stripe_upe_redirect_processed', true ) );
```
Comment (FangedParakeet) [inline tests/phpunit/test-class-wc-stripe-upe-payment-gateway.php:246]: I thought it looked nicer to use `assertTrue` here, but unfortunately even though the value that gets saved to the meta field is `true`, it gets stored as a truthy `1` instead (thus, the boolean cast is required). `assertEquals` does not perform strict type checking apparently, which is why the previous test also works.

**Review by bborman22 — APPROVED** [review]
These are great tests to get that additional coverage! They all passed for me and the tests looked good!

LGTM!
